static/js/common/upload-addon.js

```diff
@@ -261,13 +262,99 @@
             });
 
 
+
+
+
+            $('#id_admin_override_validation').addClass('addon-upload-failure-dependant')
```

Comment:
Most of the code comes from `dehub.js` yes. I just factored out the `isUnlisted` and `isSideload` detection (and made them into a function).
Your question about "will it be missed there" is difficult to answer: I don't know why specifically this code related to the addon uploader wasn't in the `upload-addon.js` file to start with. It seems better to have it here, and also allows us to stay DRY.

Now, it's entirely possible that somebody will break because of this change, sadly. And as we don't have front-end tests... I guess we'll "just" need to nicely ask the QA team to be extra cautious about it?